content/200-concepts/100-components/02-prisma-client/056-composite-types.mdx

```diff
@@ -687,3 +705,50 @@ const orders = await prisma.order.findMany({
   },
 })
 ```
+
+## Duplicate values in unique fields of composite types
+
+Be careful when you carry out any of the following operations on a record with a composite type that has a unique constraint. In this situation, MongoDB does not enforce unique values inside a record.
+
+- When you create the record
+- When you add data to the record
+- When you update data in the record
+
+If your schema has a composite type with a `@@unique` constraint, MongoDB prevents you from storing the same value for the constrained value in two or more of the records that contain this composite type. However, MongoDB does does not prevent you from storing multiple copies of the same field value in a single record. For example, in the following schema, `MailBox` has a composite type, `addresses`, which has a `@@unique` constraint on the `email` field.
+
+```prisma
+type Address {
+  email String
+}
+
+model MailBox {
+  name       String
+  addresses  Address[]
+
+  @@unique([addresses.email])
+}
+```
+
+The following code creates a record with two identical values in `address`. MongoDB does not throw an error in this situation, and it stores `alice@prisma.io` in `addresses` twice.
+
+```ts
+await prisma.MailBox.createMany({
+  data: [
+    {
+      name: ‘Alice’,
+      addresses: {
+        set: [
+          {
+            address: ‘alice@prisma.io‘,
+          },
+          {
+            address: 'alice@prisma.io‘',
+          },
+        ],
+      },
+    },
+  ],
+})
+```
+
+Note that MongoDB does throw an error if you try to store the same value in two separate records. In our example above, if you try to store the email address `alice@prisma.io` for the user Alice and for the user Bob, MongoDB does not store the data and throws an error.
```

Comment:
Wondering if its worth nothing how to do this if you wish to enforce the address to be unique. You would have to define two models: 

```gql
model Address {
    id        String   @id @default(auto()) @map("_id") @db.ObjectId
    name      String
    mailbox   Mailbox? @relation(fields: [mailboxId], references: [id])
    mailboxId String?  @db.ObjectId

    @@unique([name])
}

model Mailbox {
    id        String    @id @default(auto()) @map("_id") @db.ObjectId
    name      String
    addresses Address[] @relation()
}
```

Then if you were to create two nested records with the same address: 

```ts
await prisma.MailBox.create({
  data: {
    name: 'Alice',
    addresses: {
      create: [
        { name: 'alice@prisma.io' }, 
        { name: 'alice@prisma.io' }
      ],
    },
  },
})
```

It would fail thus enforcing the unique constraint.

Reply:
Thanks - yes, I think it's worth including this workaround

Reply:
I've brought this workaround in, so I'm closing the comment thread